pulp_smash/tests/pulp2/rpm/api_v2/test_upload_publish.py

```diff
@@ -468,3 +469,47 @@ def _verify_search_results(self, units):
             units[0]['metadata']['vendor'],
             RPM_WITH_VENDOR_DATA['metadata']['vendor'],
         )
+
+
+class UploadInvalidRPMTestCase(unittest.TestCase):
+    """Test that upload fails and error is raised when upload invalid RPM."""
+
+    def test_all(self):
+        """Test whether one invalid RPM upload fails and produces error details.
+
+        This test targets the following issues.
+
+        * `Pulp Smash #544 <https://github.com/PulpQE/pulp-smash/issues/544>`_
+        * `Pulp #2543 <https://pulp.plan.io/issues/2543>`_
+        * `Pulp #3090 <https://pulp.plan.io/issues/3090>`_
+
+        Do the following:
+
+        1. Create a RPM repository.
+        2. Upload an invalid RPM to repository.
+        3. Assert that error message is descriptive of the action that
+           triggered the error.
+        4. Assert that upload fails.
```

Comment:
The code in step 4 doesn't actually assert that the upload fails. That's what steps 2 and 3 do. Instead, the code in step 4 searches verifies that the repository contains no RPMs. I'd suggest writing something like this instead:

> Verify that the repository contains no RPMs.